app/src/main/java/com/github/geohunt/app/ComposeActivity.kt

```diff
@@ -0,0 +1,121 @@
+package com.github.geohunt.app
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import com.github.geohunt.app.event.marker.EventMarkerActionDisplay
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import java.time.LocalDateTime
+import java.time.Month
+
+class ComposeActivity : AppCompatActivity(), OnMapReadyCallback {
+    private lateinit var map: GoogleMap
+    //Hardcoded list used to test correct display of events on the map
+    private val REQUEST_CODE_LOCATION_PERMISSION = 1
+    private var mockBitmap: Bitmap = Bitmap.createBitmap(IntArray(120*120){Color.CYAN}, 90, 90, Bitmap.Config.ARGB_8888)
+    private var mockChallengeDatabase : List<Triple<Bitmap, LatLng, LocalDateTime>> = listOf(
+            Triple(mockBitmap, LatLng(46.51958, 6.56398), LocalDateTime.of(2023, Month.MAY, 1, 19, 39, 12)),
+            Triple(mockBitmap, LatLng(46.52064, 6.56780), LocalDateTime.of(2023, Month.MAY, 2, 12, 24, 35)),
+            Triple(mockBitmap, LatLng(46.51881, 6.56779), LocalDateTime.of(2023, Month.MAY, 3, 16, 12, 12)))
```

Comment:
This probably should be injected as a dependency for model/view separation, and easier testing by the way.

Reply:
On another note, creating a special data class to hold challenge information instead of a `Triple` would probably make it clearer to use.

Reply:
I have implemented a new class to represent the data of the marker pin. This new class allows us to store the challenge information in a more structured and organized way than using a Triple. It also makes the code more readable and easier to maintain, as the logic related to the marker data is encapsulated in a separate class.